pypeit/core/datacube.py

```diff
@@ -126,8 +257,63 @@ def dar_correction(wave_arr, coord, obstime, location, pressure, temperature, re
     return ra_diff, dec_diff
 
 
+def make_whitelight_fromref(all_ra, all_dec, all_wave, all_sci, all_wghts, all_idx, dspat,
+                            ref_filename, numfiles=None):
+    """ Generate a whitelight image of every input frame,
+    based on a reference image. Note the, the reference
+    image must have a well-defined WCS.
+
+    Args:
+        all_ra (`numpy.ndarray`_):
+            RA values of each pixel from all spec2d files
+        all_dec (`numpy.ndarray`_):
+            DEC values of each pixel from all spec2d files
+        all_wave (`numpy.ndarray`_):
+            Wavelength values of each pixel from all spec2d files
+        all_sci (`numpy.ndarray`_):
+            Counts of each pixel from all spec2d files
+        all_wghts (`numpy.ndarray`_):
+            Weights attributed to each pixel from all spec2d files
+        all_idx (`numpy.ndarray`_):
+            An integer identifier indicating which spec2d file each pixel originates from.
+            For example, a 0 would indicate that a pixel originates from the first spec2d
+            frame listed in the input file. a 1 would indicate that this pixel originates
+            from the second spec2d file, and so forth.
+        dspat (float):
+            The size of each spaxel on the sky (in degrees)
+        ref_filename (str):
+            A fits filename of a reference image to be used when generating white light
+            images. Note, the fits file must have a valid 3D WCS.
+        numfiles (int, optional):
```

Comment:
Omit redundant arguments?

Reply:
I'm not sure what you mean by redundant, here. `numfiles` is passed to  `make_whitelight`

Reply:
It is just the size of all_idx, so just determine if in make_whitelight from that array, unless I'm missing something?